src/main/java/net/snowflake/client/loader/ProcessQueue.java

```diff
@@ -144,12 +146,12 @@ public void run() {
 
           while (rs.next()) {
             // Get the number of rows actually loaded
-            loaded += rs.getInt("rows_loaded");
+            loaded += rs.getLong("rows_loaded");
             // Get the number of rows parsed
-            parsed += rs.getInt("rows_parsed");
+            parsed += rs.getLong("rows_parsed");
           }
 
-          int errorRecordCount = parsed - loaded;
+          int errorRecordCount = toIntExact(parsed - loaded);
```

Comment:
could this be `long`?

Reply:
This value is used for the function LoadResultListener#getErrorRecordCount() and LoadResultListener#addErrorRecordCount(). So the signature of these functions would change from int to long. This would be a behavior change. I don't think that change would be worth it since there have been no complaints about processQueue and it might break existing customers. So I'd prefer to keep these signature as ints, and throw an exception if the long can't be converted. What do you think?

Reply:
Disregard; adding both methods.

Reply:
Yes, I agree. You could just cast the `long` value to `int` and let the overflow without error. The customers should have already gotten a wrong value anyway. Raising an exception will be a behavior change.

Reply:
This processQueue change would require restructuring a large number of internal tests. I don't think this is worth it for something that's never generated any complaints. Reverting back to previous working version.

Reply:
So do we still want to throw an exception when overflow occurs?